tests/lax_scipy_sparse_test.py

```diff
@@ -60,7 +61,34 @@ def rand_sym_pos_def(rng, shape, dtype):
   return matrix @ matrix.T.conj()
 
 
+
+
+
+
 class LaxBackedScipyTests(jtu.JaxTestCase):
+  def _fetch_preconditioner(self, preconditioner, A, rng=None,
+                            return_function=False):
+    """
+    Returns one of various preconditioning matrices depending on the identifier
+    `preconditioner' and the input matrix A whose inverse it supposedly
+    approximates.
+    """
+    if preconditioner == 'identity':
+      M = np.eye(A.shape[0], dtype=A.dtype)
+    elif preconditioner == 'random':
+      if rng is None:
+        rng = jtu.rand_default(self.rng())
+      M = np.linalg.inv(rand_sym_pos_def(rng, A.shape, A.dtype))
+    elif preconditioner == 'exact':
+      M = np.linalg.inv(A)
+    else:
+      M = None
+
+    if M is None or not return_function:
+      return M
+    else:
+      return lambda x: jnp.dot(M, x, precision=lax.Precision.HIGHEST)
```

Comment:
Why is the highest precision needed here? The preconditioner is often an approximation of the inverse of A anyway.

Reply:
Ah, I see this is just for testing. Fair enough.

Reply:
We do use the highest precision throughout though, so it's a fair point